tensorboard/webapp/widgets/data_table/types.ts

```diff
@@ -42,6 +43,44 @@ export enum ColumnHeaderType {
   CUSTOM = 'CUSTOM',
 }
 
+export enum DomainType {
+  DISCRETE,
+  INTERVAL,
+}
+
+export type DiscreteFilterValues = string[] | number[] | boolean[];
+
+export type DiscreteFilterValue = DiscreteFilterValues[number];
+
+export interface DiscreteFilter {
+  type: DomainType.DISCRETE;
+  includeUndefined: boolean;
+  possibleValues: DiscreteFilterValues;
+  // Subset of `possibleValues`
+  filterValues: DiscreteFilterValues;
+}
+
+export interface DiscreteFilterChange {
```

Comment:
I do not see this type used anywhere in this PR. Can we add it when needed? (And, by the way, it includes a field "hparamName" so you may want to figure out how to make that more generic).

Reply:
Removed